feat: improve sign up page design/layout

devChary opened the pull request.

davehakkens commented:
Thanks for helping out @devChary :)
Could you add some screenshot and descripton of the things you've changed to this PR?

devChary commented:
> Thanks for helping out @devChary :) Could you add some screenshot and descripton of the things you've changed to this PR?

devChary commented:
<img width="1728" alt="Screenshot 2024-02-19 at 12 25 12 AM" src="https://github.com/ONEARMY/community-platform/assets/26999371/1827256f-e09a-4791-87c3-43e41b48a332">

<img width="1728" alt="Screenshot 2024-02-19 at 12 25 33 AM" src="https://github.com/ONEARMY/community-platform/assets/26999371/a7e27787-8ce1-4b66-a92e-2018e406dbd6">

Here ya go with the screenshots  @davehakkens .
Tasks completed
  - Move 'already have an account' to the top, the link colour is the community brand colour and underlined
  - Each label can have a comment, which is smaller and slightly lighter than the main label
  - Consistent placeholders
  
 "Move errors to the top" -  Didn't really understand what this means so couldn't resolve it. 
Please let me know if I missed out on something.

davehakkens commented:
Thanks for adding.much better!
However in the process you seem to have closed the PR. Could you re-open so we can review it?

devChary commented:
> Thanks for adding.much better! However in the process you seem to have closed the PR. Could you re-open so we can review it?

Oh shucks! Done.

benfurber commented on src/pages/SignUp/SignUp.tsx:
```diff
@@ -132,25 +134,41 @@ const SignUpPage = observer(() => {
+                            textDecoration: 'underline',
+                          }}
+                        >
+                          {' '}
```
I don't believe we need this text spacing (`{' '}`) here.

devChary commented:
Thanks for the feedback @benfurber! 

I'll remove the custom stylings in redundant places and raise a PR with fixes for the specific comments mentioned.

benfurber commented:
@devChary Have you pushed your most recent commit?

benfurber commented:
Hey @devChary, will you have time to add the last bits to this or should I finish it off for you?

devChary commented:
Hey @benfurber, will wrap up the remaining issues by tonight and push the changes.

benfurber commented:
How are you getting on @devChary?

devChary commented:
> Hey @devChary. Thanks so much for starting to get involved!
> 
> A couple of specific comments left. More broadly, I'm looking through and I'm not convinced any of the fields should need custom stylings, spaces, etc. unless absolutely needed.
> 
> I'm thinking that the gap between fields should be set by the parent flex element, everything should stretch automatically to fill the width so no widths 100% should be needed, etc.
> 
> I'd love to use this PR as a chance to really minimise the unnecessary stylings in this component, are you up for that refactor?
> 
> On the form error stuff, please move the error rendering to appear just before the username field.
> 
> Thanks!

Hey @benfurber , 
Updates on the pending PR
I moved the error rendering to the top and removed some redundant styles.
Couple of things from your comments here. 

1. As the parent container is flex and if even if row-width is set to 100% the child (Button) loses its explicit width definition. In this case, the button's width will be determined by its content. 
2. Kept the space before `Sign-In here`, because if removed it feels to close to `Already have an account ?` which feels weird.

Open to feedback. 
I am 100% available today so any changes needed, I'll try to do asap.

benfurber commented:
Amazing! Thank you @devChary! What would you like to pick up next? :)

benfurber commented:
@all-contributors please add @devChary for code

devChary commented:
> Amazing! Thank you @devChary! What would you like to pick up next? :)

Hey @benfurber 🚀

Thanks a ton for merging the PR! 🎉  Appreciate your feedback throughout the process.
I'm currently in the midst of a job hunt 🕵️‍♂️, but rest assured, I'll get back to contributing soon! 

Cheers,
@devChary
